elyra/pipeline/component_parser_kfp.py

```diff
@@ -63,42 +65,67 @@ def _parse_properties(self, component_yaml: Dict[str, Any]) -> List[ComponentPar
         # properties.extend(self.get_runtime_specific_properties())
 
         # Then loop through and create custom properties
+        # Get parameter sub-dictionaries from YAML object
+
         input_params = component_yaml.get('inputs', [])
-        for param in input_params:
-            # KFP components default to being required unless otherwise stated.
-            # Reference: https://www.kubeflow.org/docs/components/pipelines/reference/component-spec/#interface
-            required = True
-            if "optional" in param and param.get('optional') is True:
-                required = False
-
-            # Assign type, default to string
-            data_type = param.get('type', 'string')
-
-            # Set description and include parsed type information
-            description = self._format_description(description=param.get('description', ''),
-                                                   data_type=data_type)
-
-            # Change type to reflect the type of input (inputValue vs inputPath)
-            data_type = self._get_adjusted_parameter_fields(component_body=component_yaml,
-                                                            io_object_name=param.get('name'),
-                                                            io_object_type="input",
-                                                            parameter_type=data_type)
-
-            data_type, control_id, default_value = self.determine_type_information(data_type)
-
-            # Get value if provided
-            value = param.get('default', '')
-
-            ref = param.get('name').lower().replace(' ', '_')
-            properties.append(ComponentParameter(id=ref,
-                                                 name=param.get('name'),
-                                                 data_type=data_type,
-                                                 value=(value or default_value),
-                                                 description=description,
-                                                 control_id=control_id,
-                                                 required=required))
+        output_params = component_yaml.get('outputs', [])
+
+        all_params = {"inputs": input_params, "outputs": output_params}
+
+        # Loop through inputs and outputs and create custom properties
+        for param_type, params in all_params.items():
+            for param in params:
+                # KFP components default to being required unless otherwise stated.
+                # Reference: https://www.kubeflow.org/docs/components/pipelines/reference/component-spec/#interface
+                required = True
+                if "optional" in param and param.get('optional') is True:
```

Comment:
This clause is unnecessary since `None` will be returned if the key doesn't exist and `None` != `True`.
```suggestion
                if param.get('optional') is True:
```